dashboard ui

Jsarda42 opened the pull request:
created the dashboard ui added the blackand white and dashboardcard component for flexibility

Commit:
created the dashboard ui added the blackand white and dashboardcard component for flexibility

pulgamecanica commented on src/pages/Dashboard.tsx:
```diff
@@ -1,54 +1,42 @@
+        <DashboardCard
+          title="Profile"
+          icon={<User className="text-blue-400" />}
+          to="/profil"
```
should be `profile`

pulgamecanica commented on src/pages/Dashboard.tsx:
```diff
@@ -1,54 +1,42 @@
+        <DashboardCard
+          title="Log Out"
+          icon={<LogOut className="text-blue-400" />}
+          to="/login"
```
Should probably have an action instead, which logs out, we can also create a route that logs-out the user `logout`

pulgamecanica reviewed (changes requested):
Looks, good, just need to create the logout logic and fix the `profile` typo

Would be great to already have the user hook on the Protected routes, and if a user cannot be fetched, maybe show "Session expired, please login again"

Commit:
fix profile type

Commit:
fix logout in will detail in pr

Jsarda42 commented:
# Commit Summary:
Implemented login/logout functionality with proper state management.
AuthProvider now manages authentication state (isAuthenticated, token) and handles login/logout.
AuthContext provides authentication data and functions (login, logout, isAuthenticated, etc.) to the app.
## LoginPage:
- Handles user login, displays errors, and redirects to the dashboard if authentication is successful.
## Dashboard: 
- Displays a logout button that triggers the logout function and redirects the user to the login page.
## Key Changes:
- Updated AuthProvider to manage isAuthenticated state and handle login/logout.
- Integrated AuthContext and useAuth hook to access auth data in components.
- Added a logout button in the Dashboard that triggers the logout functionality.

Commit:
expired sessions detail in pr

pulgamecanica commented on src/context/AuthProvider.tsx:
```diff
@@ -27,12 +29,14 @@ export function AuthProvider({ children }: { children: ReactNode }) {
     const newToken = await loginUser({ username, password });
```
Probably should be inside an `else`

pulgamecanica commented:
Great job! It works really well.
